File: C_string/rs/src/c_string/c2.rs
// Tulis function analyzeCharacters di sini
#[allow(dead_code)]
#[derive(Debug)]
pub struct Character {
    char: char,
    position: usize,
}

pub fn analyze_characters(text: &str) -> Vec<Character> {
    let mut result: Vec<Character> = Vec::new();

    for (i, c) in text.chars().enumerate() {
        result.push(Character {
            char: c,
            position: i,
        });
    }
    result
}

File: projects/experimental/challenge/C_string/ts/c2.ts
// Tulis function analyzeCharacters di sini
interface Character {
    char: string,
    position: number

};
type characterfn = (text: string) => Character[];

const analyzeCharacters: characterfn = (text: string): Character[] => {
    const result: Character[] = [];

    for (let i = 0; i < text.length; i++) {
        result.push({
            char: text.charAt(i),
            position: i
        });
    }
    return result
}
console.log(analyzeCharacters("Hi"));
console.log(analyzeCharacters("Code"));